examples/funsAndScopes.browse

```diff
@@ -0,0 +1,127 @@
+# Create a scope
+scope math {
+  vars {
+    PI 3.14
+    e 2.718
+  }
+  private { } # a.k.a internal
+  rules {
+    abs {
+      bind $x
+      # TODO: somehow eval Math.abs(x) in javascript and return that
+      return $x
+    }
+  }
+}
+
+##### OPTION 1 #####
+# We can extend the match scope directly and create additional rules that are
+# available whenever we're in a match scope
+extend math {
+  rules {
+    # Not the real implementation we use because Math.pow in JS is far
+    # more efficient but this serves as a good example, that's all
+    pow {
+      bind x y
+      if ($y == 0) then { return 1 }
+      
+      # We can even use the `abs` rule that's only available in the math scope
+      set recursed ( pow $x ( abs $y / 2 ) )
+
+      if ($y % 2 == 0) then {
+        return $recursed * $recursed
+      } else {
+        set v 
+        return $x * $recursed * $recursed
+      }
+    }
+  }
+}
+
+using math {
+  pow 2 8
+}
+
+#### OPTION 2 #####
+# Instead of extending a scope, which feels quite OOP, rules can do 2 new
+# things: 
+#   1. They can define which scope they are accessible in. This way, they still
+#      need to be independently imported, passed around, etc. However, they can
+#      only be evaluated in a context where the page scope is an ancestor. 
```

Comment:
Do you mean "Where the scope they are accessible in is an ancestor?" or am I misunderstanding

Reply:
yup, you're right. I changed the comments a few times and this line looks like it belonged to an old example